pyproject.toml

```
@@ -42,7 +42,7 @@ pandas = "2.3.3"
 pyarrow = ">=10.0.1"
 pytest = ">=8.4.2"
 pyright = ">=1.1.407"
-ty = ">=0.0.1a24"
+ty = "0.0.1a25"
```

Newer version has been released
```suggestion
ty = ">=0.0.1a26"
```

---

> I may add more ignore and report to ty project, it’s causing some issues with union type

My take on any type checker is that if they create a new version that causes our type checking to fail, and the others are still happy with the code, we pin the version of the "failing" type checker, and create a simple case and report it to the type checker's maintainers.

So in this case, you pin the version that works for `ty`, and hopefully create a simple case that illustrates the issue for them.